src/backend/distributed/planner/distributed_planner.c

```diff
@@ -2474,6 +2513,18 @@ GetRTEListProperties(List *rangeTableList)
 		else if (IsCitusTableTypeCacheEntry(cacheEntry, DISTRIBUTED_TABLE))
 		{
 			rteListProperties->hasDistributedTable = true;
+
+			if (IsCitusTableTypeCacheEntry(cacheEntry, NULL_KEY_DISTRIBUTED_TABLE))
+			{
+				rteListProperties->nullDistKeyTableColocationIds = list_append_unique_oid(
```

Comment:
is there a more generic solution? e.g. what would we do if we needed to solve this problem for non-co-located router queries?